docs/howtos/layers/surface.md

```diff
@@ -0,0 +1,122 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.10.3
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# Using the surface layer
+
+In this document, you will learn about the `napari` `Surface` layer, including
+how to display surface data and edit the properties of surfaces like the
+contrast, opacity, colormaps and blending mode. You will also understand how to
+add and manipulate surfaces both from the GUI and from the console.
+
+## When to use the surface layer
+
+The surface layer allows you to display a precomputed surface mesh that is
+defined by an `NxD` array of `N` vertices in `D` coordinates, an `Mx3` integer
+array of the indices of the triangles making up the faces of the surface, and a
+length `N` list of values to associate with each vertex to use alongside a
+colormap.
+
+## A simple example
+
+You can create a new viewer and add a surface in one go using the
+`napari.view_surface` method, or if you already have an existing viewer, you can
+add an image to it using `viewer.add_surface`. The api of both methods is the
+same. In these examples we'll mainly use `view_surface`.
+
+A simple example of viewing a surface is as follows:
+
+```{code-cell} python
+import napari
+import numpy as np
+
+vertices = np.array([[0, 0], [0, 20], [10, 0], [10, 10]])
+faces = np.array([[0, 1, 2], [1, 2, 3]])
+values = np.linspace(0, 1, len(vertices))
+surface = (vertices, faces, values)
+
+viewer = napari.view_surface(surface)  # add the surface
+```
+
+```{code-cell} python
+:tags: [hide-input]
+
+from napari.utils import nbscreenshot
+
+nbscreenshot(viewer)
+```
+
+```{code-cell} python
+:tags: [remove-cell]
+
+viewer.close()
+```
+
+## Arguments of `view_surface` and `add_surface`
+
+{meth}`~napari.view_layers.view_surface` and {meth}`~napari.Viewer.add_surface`
+accept the same layer-creation parameters.
+
+```{code-cell} python
+:tags: [hide-cell]
+
+help(napari.view_surface)
+```
+
+## Surface data
+
+The data for a surface layer is defined by a 3-tuple of its vertices, faces, and
+vertex values. The vertices are an `NxD` array of `N` vertices in `D`
+coordinates. The faces are an `Mx3` integer array of the indices of the
+triangles making up the faces of the surface. The vertex values are a length `N`
+list of values to associate with each vertex to use alongside a colormap. This
+3-tuple is accessible through the `layer.data` property.
```

Comment:
I know this is just moving content around,  so maybe that should be another issue.

Do you know if there are any ordering of vertices for faces in order to cull hidden faces ? Or self-intersection ? I have issues with "Suzanne" from blender. 

<img width="664" alt="Screen Shot 2022-02-21 at 17 48 53" src="https://user-images.githubusercontent.com/335567/154997516-160af393-987b-422a-956a-2b8e472a2f8a.png">

But arguably, maybe my code is wrong:

```python
from stl import mesh
import napari
import numpy as np
from os.path import expanduser
# https://fr.wikipedia.org/wiki/Fichier:Suzanne.stl
m = mesh.Mesh.from_file(expanduser('~/Downloads/Suzanne.stl'))
# make suzane face us
m.rotate((0,0,1), np.pi/2)
m.rotate((1,0,0), -np.pi/2)
vs = list(m.vectors.reshape(-1, 1, 3).squeeze(1))

vertices = np.array(sorted({tuple(x) for x in vs}))

faces = []
for face in m.vectors:
    fc = []
    for vec in face:
        wh = np.where((vertices == np.array(vec)).sum(axis=1) == 3)
        idx = wh[0][0]
        fc.append(idx)
    faces.append(fc)
faces = np.array(faces)    

faces = np.array(faces)
values = np.linspace(0, 1, len(vertices))
surface = (vertices, faces, values)
viewer = napari.view_surface(surface)  # add the surface
```

Reply:
I don't know, but this does feel like another issue. I think there are some gl settings under the hood for culling that we can change around if necessary

Reply:
#4135